make/generate.mk

```
@@ -18,15 +18,32 @@ generate: generate-deepcopy-and-crds generate-openapi dispatch-crds copy-reg-ser
 generate-deepcopy-and-crds: remove-config controller-gen
 	@echo "Re-generating the deepcopy go file & the Toolchain CRD files... "
 	$(Q)$(CONTROLLER_GEN) crd \
-	object paths="./..." output:crd:artifacts:config=config/crd/bases
+	object paths="./..." output:crd:artifacts:config=$(PATH_TO_CRD_BASES)
 	
 .PHONY: generate-openapi
 generate-openapi: openapi-gen
 	@echo "re-generating the openapi go file..."
-	$(Q)$(OPENAPI_GEN) --input-dirs ./api/$(API_VERSION)/ \
+	@## openapi-gen requires the GOPATH env var be set and the codebase be present within it.
+	@## Let's not require $GOPATH be set up in the user's environment and the checkout be
+	@## placed in it.
+	@## Instead, fake it locally.
+	mkdir -p $(PROJECT_DIR)/.fake-gopath
+	@mkdir -p $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain
+	@## link the packages from the real GOPATH
+	@## ("go env GOPATH" returns the correct value even if the env var is not set)
+	@cd $(PROJECT_DIR)/.fake-gopath && ln -s `go env GOPATH`/pkg || true
+	@## link our codebase to the appropriate place in the fake GOPATH
+	@cd $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain && ln -s ../../../.. api
+	@## run openapi-gen from within the fake GOPATH (otherwise the package paths would be relative
+	@#e and function names would be different)
+	export GOPATH=$(PROJECT_DIR)/.fake-gopath \
+	&& cd $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain/api \
+	&& $(Q)$(OPENAPI_GEN) --input-dirs ./api/$(API_VERSION)/ \
```

I guess that we don't need to export the variable, right? this should be enough:
```suggestion
	&& cd $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain/api \
	&& GOPATH=$(PROJECT_DIR)/.fake-gopath $(Q)$(OPENAPI_GEN) --input-dirs ./api/$(API_VERSION)/ \
```

---

btw, we can also drop the `$(Q)` - no clue why it is there
```suggestion
	&& cd $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain/api \
	&& GOPATH=$(PROJECT_DIR)/.fake-gopath $(OPENAPI_GEN) --input-dirs ./api/$(API_VERSION)/ \
```